packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-custom-pwd-recovery-mfa/main/aspnet/sdk.md

```diff
@@ -0,0 +1,75 @@
+### 1: Initiate password recovery
+
+To begin the password recovery flow, the user clicks the **Forgot Password** button on the login page. Then they select **Email** as the authenticator to be used for password recovery on the subsequent page and click **Submit**. Okta then sends an email to their email address matching the Forgot Password template that was altered earlier.
+
+<div class="common-image-format">
+
+![Screenshot of email sent to user](/img/advanced-use-cases/custom-pwd-recovery-custom-email.png)
+
+</div>
+
+The Reset Password link in the email includes the `OTP` and `request.relayState` variables as query parameters back to the application. For example,
+
+`https://localhost:44314/magiclink/callback?otp=${oneTimePassword}&state=${request.relayState}` becomes `https://localhost:44314/magiclink/callback?otp=726009&state=1b34371af02dd31d2bc4c48a3607cd32`.
+
+### 2: Handle the otp and state parameters
+
+Create a callback handler method that takes the `otp` and `state` parameters in the query string and passes them as parameters to the `VerifyAuthenticatorAsync` method on the `IdxClient`.
+
+> **Note** : This implementation looks for the idx context in session, keyed either by state or by the string `idxContext`. Alternatively, you may choose to store the idx context in a database or a file.
```

Comment:
```suggestion
> **Note** : This implementation looks for the IDX context in the session, keyed either by state or by the string `idxContext`. Alternatively, you may choose to store the IDX context in a database or a file.
```

Reply:
Changed to This implementation looks for the current IDX context in a session variable keyed either by the value of `state` or by the string `idxContext`. Alternatively, you may choose to store the IDX context in a database or a file.